VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts

```diff
@@ -1125,7 +1127,7 @@ export abstract class WaitingRoomBaseDirective {
     }
 
     shouldCurrentUserJoinHearing(): boolean {
-        return !this.isHost() || this.participant.status === ParticipantStatus.InHearing;
+        return !this.isHost() || this.participant.status === ParticipantStatus.InHearing || this.shouldUpdateHostShowVideo;
```

Comment:
Is this just a check for hosts? Rather than storing this boolean in the base component could we have that in the judge-waiting-room.component.ts and override this method? Maybe not since you do set it elsewhere in this file but it would be nice to keep the host specific stuff in the specific component.

Reply:
we should call shouldCurrentUserJoinHearing in waiting-room-base.component.ts rather than in judge-waiting-room.component.ts

Reply:
I was thinking we could have a method in the judge-waiting-room.component.ts that looks like this.

`
shouldCurrentUserJoinHearing(): boolean {
    return super.shouldCurrentUserJoinHearing() || this.shouldUpdateHostShowVideo;
}
`

Just if it's only for hosts we don't end up putting more code in the base component that isn't relevant for others

Reply:
there is no anywhere calling shouldCurrentUserJoinHearing this method in  judge-waiting-room.component.ts